### PR TITLE
Add a helper to determine if a type has a specific attribute or not.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs [hunk 1]
```diff
@@ -796,4 +796,12 @@ public static bool IsObsolete(this ISymbol symbol)
             MetadataName: nameof(ObsoleteAttribute),
             ContainingNamespace.Name: nameof(System),
         });
+
+    public static bool HasAttribute([NotNullWhen(true)] this ISymbol? symbol, INamedTypeSymbol? attributeClass)
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs:800]: >NotNullWhen [](http://example.com/codeflow?start=37&length=11)

consider decorating the other parameter too

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ICompilationExtensions.cs [hunk 2]
```diff
@@ -79,7 +79,10 @@ public static ImmutableArray<IAssemblySymbol> GetReferencedAssemblySymbols(this
         => compilation.GetTypeByMetadataName(typeof(System.Diagnostics.DebuggerDisplayAttribute).FullName!);
 
     public static INamedTypeSymbol? StructLayoutAttributeType(this Compilation compilation)
-        => compilation.GetTypeByMetadataName(typeof(System.Runtime.InteropServices.StructLayoutAttribute).FullName!);
+        => compilation.GetTypeByMetadataName(typeof(StructLayoutAttribute).FullName!);
+
+    public static INamedTypeSymbol? InlineArrayAttributeType(this Compilation compilation)
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ICompilationExtensions.cs:84]: >public static INamedTypeSymbol? InlineArrayAttributeType(this Compilation compilation) [](http://example.com/codeflow?start=4&length=86)

nit: from other PR

File: src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs [hunk 3]
```diff
@@ -298,15 +298,13 @@ static bool IsInterpolatedStringHandlerMandatoryConstructorParameter(
...
+                    !(method.Parameters is [_, var secondParameter, ..] && secondParameter == parameter))
                 {
-                    if (!(method.Parameters is [_, var secondParameter, ..] && secondParameter == parameter))
-                    {
-                        return false;
-                    }
+                    return false;
                 }
 
-                if (!containingType.GetAttributes().Any(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, interpolatedStringHandlerAttributeType)))
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs:309]: No.  Either is fine

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs]: done.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ICompilationExtensions.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ICompilationExtensions.cs]: done.

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**